src/data_structures/fmindex.rs

```diff
@@ -168,18 +183,46 @@ impl FMDIndex {
     ///   less memory usage, but worse performance)
     pub fn new(bwt: BWT, k: usize) -> Self {
         let mut alphabet = dna::n_alphabet();
-        alphabet.insert(b'$');
-        assert!(
-            alphabet.is_word(&bwt),
-            "Expecting BWT over the DNA alphabet (including N) with the sentinel $."
-        );
+        alphabet.insert(SENT_CHAR);
+        assert!(alphabet.is_word(&bwt),
+                "Expecting BWT over the DNA alphabet (including N) with the sentinel $.");
 
         FMDIndex {
             fmindex: FMIndex::new(bwt, k, &alphabet),
-            revcomp: dna::RevComp::new()
+            revcomp: dna::RevComp::new(),
         }
     }
 
+    /// Find longest non-encompassing prefix match
+    pub fn longest_prefix_match(&self, pattern: &[u8], min_overlap: usize) -> Option<BiInterval> {
+        let exact_match_interval = self.fmindex.backward_search(pattern.iter());
```

Comment:
It seems strange that the behavior of an FMIndex is exclusively backward_search, and smems for FMDIndex. There are many more algorithms that exist for these/this data structure/s, so I'm not sure why the underlying functionality (occ, less, init, forward/backward extend) is encapsulated without also providing these algorithms. Essentially, it makes it so that client developers (like me :) ) have to change the underlying library code to get the algorithm they want. Perhaps the eventual goal is to have inbuilt support for all common FMIndex/FMDIndex algorithms, so that clients won't have to implement them?

Anyway. This is definitely getting out of scope for this particular PR. Thanks for your help-- I'll get something together for a more by-the-book algorithm soon.